### PR TITLE
ROX-21880: Compliance cluster results table

### PR DESCRIPTION
**Pull request by bradr5**
## Description

This adds the table for viewing cluster compliance check results by scan configuration

Notes:

- Mocks show the compliance checks filtering as a dropdown with prefilled checks, however, we don't have an endpoint to get the list of checks currently. So we went with a free form input field instead for now

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Default table view:
Screenshot 2024-01-31 at 4 38 59 PM

Filtered:
Screenshot 2024-01-31 at 4 39 30 PM

### CODE REVIEW

File: ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/Coverage/Clusters/ClusterDetailsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+    ComplianceCheckStatus,
+    ClusterCheckStatus,
+    getSingleClusterResultsByScanConfig,
+    getSingleClusterResultsByScanConfigCount,
+} from 'services/ComplianceEnhancedService';
+import { SearchFilter } from 'types/search';
+import { SortOption } from 'types/table';
+import { searchValueAsArray } from 'utils/searchUtils';
+
+import TableErrorComponent from 'Containers/Vulnerabilities/WorkloadCves/components/TableErrorComponent';
```
Comment (vjwilson) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/Coverage/Clusters/ClusterDetailsTable.tsx:34]: Let's add a `TODO` comment to remind us to abstract this out of Vulnerabilities folder

File: ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/Coverage/Clusters/ClusterDetailsTable.tsx [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+import { SearchIcon } from '@patternfly/react-icons';
```
Comment (vjwilson) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/Coverage/Clusters/ClusterDetailsTable.tsx:35]: Move this line up to the top section where the other 3rd party imports are.

File: ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/Coverage/Clusters/ClusterDetailsTable.tsx [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+    const { data: scanResults, loading: isLoading, error } = useRestQuery(listQuery);
+
+    const countQuery = useCallback(
+        () => getSingleClusterResultsByScanConfigCount(clusterId, scanName, searchFilter),
+        [clusterId, scanName, searchFilter]
+    );
+    const { data: scanResultsCount } = useRestQuery(countQuery);
+
+    function getStatusByClusterId(clusters: ClusterCheckStatus[]): ComplianceCheckStatus | null {
+        const cluster = clusters.find((cluster) => cluster.cluster.clusterId === clusterId);
```
Comment (vjwilson) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/Coverage/Clusters/ClusterDetailsTable.tsx:84]: "Cluster! Cluster! Cluster ID!"
(Jan Brady)

File: ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/Coverage/Clusters/ClusterDetailsTable.tsx [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+            const status = getStatusByClusterId(clusters);
+            const statusObj = status ? getClusterResultsStatusObject(status) : null;
+
+            return (
+                <Tr key={checkName}>
+                    <Td modifier="truncate">
+                        <>
+                            <Link to={scanConfigUrl}>{checkName}</Link>
+                            <br />
+                            <small>{rationale}</small>
```
Comment (vjwilson) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/Coverage/Clusters/ClusterDetailsTable.tsx:135]: for a TODO: It seems like there is a PatternFly component for this look, with main info larger on top, and further info smaller on the bottom.

File: ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/Coverage/Components/CheckStatusDropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    function onCveSeverityToggle(isOpen: boolean) {
+        setCheckStatusIsOpen(isOpen);
+    }
+
+    return (
+        <Select
+            variant="checkbox"
+            aria-label="CVE severity filter menu items"
+            toggleAriaLabel="CVE severity filter menu toggle"
```
Comment (vjwilson) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/Coverage/Components/CheckStatusDropdown.tsx:22]: copy/paste miss: need to update these strings for Compliance status

**Review by vjwilson — CHANGES_REQUESTED** [review]
There are a couple of errors that need to be fixed when you push that bug change, and a couple of TODOs to add.

File: ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/Coverage/Clusters/ClusterDetailsTable.tsx (hunk 3 above)
Comment (bradr5) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/Coverage/Clusters/ClusterDetailsTable.tsx:84]: 🤣

File: ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/Coverage/Clusters/ClusterDetailsTable.tsx (hunk 4 above)
Comment (bradr5) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/Coverage/Clusters/ClusterDetailsTable.tsx:135]: Hmm any idea where they have that? I couldn't find a devoted component so I ended up mimicking a pattern in one of the table examples [here](https://v4-archive.patternfly.org/v4/components/table#composable-nested-column-headers). Although the example is just noting the striped rows and doesn't appear to be intentionally included in the component. They also have [this](https://v4-archive.patternfly.org/v4/components/table#composable-nested-column-headers) which includes faded text but not small.

**Review by vjwilson — APPROVED** [review]

**Pull request merged**